arwen/host/execution.go

```diff
@@ -760,13 +761,68 @@ func (host *vmHost) callBuiltinFunction(input *vmcommon.ContractCallInput) (*vmc
 		for _, outAcc := range vmOutput.OutputAccounts {
 			outAcc.OutputTransfers = make([]vmcommon.OutputTransfer, 0)
 		}
+	} else {
 	}
 
 	output.AddToActiveState(vmOutput)
 
 	return newVMInput, gasConsumedForExecution, nil
 }
 
+// add output transfer of esdt transfer when needed - sc calling another sc intra shard without extra call
+func (host *vmHost) addESDTTransferToVMOutputSCIntraShardCall(
```

Comment:
Function does not appear to be called anywhere.

Reply:
called it.